test/io/serializers/warn-interfaces.chpl

```diff
@@ -0,0 +1,278 @@
+    writer.write(x);
+  }
+
+  proc init(x = 0) { this.x = x; }
+
+  proc init(reader, ref deserializer) throws {
+    init this;
+    this.x = reader.read(int);
+  }
+}
+
+class JustTwoClass : writeSerializable, readDeserializable {
+  var x : int;
+
+  override proc serialize(writer, ref serializer) throws {
+    writer.write(x);
+  }
+
+  override proc deserialize(reader, ref deserializer) throws {
+    this.x = reader.read(int);
+  }
+
+  proc init(x = 0) {
+    this.x = x;
+  }
+
+  proc init(reader, ref deserializer) {
+    init this;
+    this.x = reader.read(int);
+  }
+}
+
+class SerializableClass : serializable {
+  var x : int;
+
+  override proc serialize(writer, ref serializer) throws {
+    writer.write(x);
+  }
+
+  override proc deserialize(reader, ref deserializer) throws {
+    this.x = reader.read(int);
+  }
+
+  proc init(x = 0) {
+    this.x = x;
+  }
+
+  proc init(reader, ref deserializer) {
+    init this;
+    this.x = reader.read(int);
+  }
+}
+
+class DefaultClass {
+  var x : int;
+}
+
+class SerializeMissingClass {
+  var x : int;
+
+  override proc serialize(writer, ref serializer) throws {
+    writer.write(x);
+  }
+}
+
+class SerializePresentClass : writeSerializable {
+  var x : int;
+
+  override proc serialize(writer, ref serializer) throws {
+    writer.write(x);
+  }
+}
+
+class DeserializeMissingClass : writeSerializable {
+  var x : int;
+
+  override proc serialize(writer, ref serializer) throws {
+    writer.write(x);
+  }
+
+  override proc deserialize(reader, ref deserializer) throws {
+    this.x = reader.read(int);
+  }
+}
+
+class DeserializePresentClass : writeSerializable, readDeserializable {
+  var x : int;
+
+  override proc serialize(writer, ref serializer) throws {
+    writer.write(x);
+  }
+
+  override proc deserialize(reader, ref deserializer) throws {
+    this.x = reader.read(int);
+  }
+}
+
+class InitMissingClass : writeSerializable {
+  var x : int;
+
+  override proc serialize(writer, ref serializer) throws {
+    writer.write(x);
+  }
+
+  proc init(x = 0) { this.x = x; }
+
+  proc init(reader, ref deserializer) throws {
+    init this;
+    this.x = reader.read(int);
+  }
+}
+
+class InitPresentClass : writeSerializable, initDeserializable {
+  var x : int;
+
+  override proc serialize(writer, ref serializer) throws {
+    writer.write(x);
+  }
+
+  proc init(x = 0) { this.x = x; }
+
+  proc init(reader, ref deserializer) throws {
+    init this;
+    this.x = reader.read(int);
+  }
+}
+
+enum mode { no, readVal, readInit }
+use mode;
+
+proc test(type T, param read = mode.readVal) {
+  writeln("----- ", T:string, " -----");
+  var f = openMemFile();
+  var val = new T(10);
+  f.writer().write(val);
+
+  if read == readVal {
+    var newVal = new T();
+    f.reader().read(newVal);
+    writeln("got: ", newVal);
+    const check = if isClassType(T) then newVal! else newVal;
+    assert(check.x == 10);
+  } else if read == readInit {
+    var newVal : T = f.reader().read(T);;
```

Comment:
extra `;`

Reply:
I'll fix, but FWIW a single ``;`` is a valid statement in Chapel, so this won't cause failures when mistyping!